src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -1574,16 +1579,35 @@ UndistributeTable(Oid relationId)
 						errdetail("because the table is not distributed")));
 	}
 
-	if (TableReferencing(relationId))
+	bool tableReferencing = TableReferencing(relationId);
+	bool tableReferenced = TableReferenced(relationId);
+	if ((tableReferencing || tableReferenced) && cascade)
```

Comment:
maybe is a little more readable if we can avoid the function calls/extra variables.

 If cascade is false, we still call TableReferencing/TableReferenced once below, so performance wise also same.
```C
if (cascade && (TableReferencing(relationId) || TableReferenced(relationId))
{
}

```